sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -189,8 +189,13 @@ class RocksDB(
         // reset last snapshot version
         if (lastSnapshotVersion > latestSnapshotVersion) {
           // discard any newer snapshots
-          lastSnapshotVersion = 0L
+          synchronized {
+            latestSnapshot = None
```

Comment:
Need to move this snapshot to oldSnapshots so that it can be cleaned up by maintenance task.

Reply:
Done